pyhealth/datasets/base_dataset.py

```diff
@@ -593,3 +603,170 @@ def task_stat(self) -> str:
+
+    def _index_visit(self) -> Dict[str, List[int]]:
+        """Helper function which indexes the samples by visit_id.
+
+        Will be called in set_task().
+
+        Returns:
+            visit_to_index: Dict[str, int], a dict mapping visit_id to a list
+                of sample indices.
+        """
+        visit_to_index = {}
+        for idx, sample in enumerate(self.samples):
+            visit_to_index.setdefault(sample["visit_id"], []).append(idx)
+        return visit_to_index
+
+
+    # TODO: check this
+    def get_all_tokens(self, key: str, sort: bool = True) -> List[str]:
+        """Gets all tokens with a specific key in the samples.
+
+        Args:
+            key: str, the key of the tokens in the samples.
+            sort: whether to sort the tokens by alphabet order.
+
+        Returns:
+            tokens: a list of tokens.
+        """
+        tokens = []
+        for sample in self.samples:
+            # for multi-class classification
+            if type(sample[key]) in [bool, int, str]:
+                tokens.append(sample[key])
+            # for multi-label classification
+            elif type(sample[key][0]) in [bool, int, str]:
+                tokens.extend(sample[key])
+            elif type(sample[key][0]) == list:
+                tokens.extend(sample[key][-1])
+            else:
+                raise ValueError(f"Unknown type of {key}: {type(sample[key])}")
+        tokens = list(set(tokens))
+        if sort:
+            tokens.sort()
+        return tokens
+
+    # TODO: check this
+    def get_label_distribution(self) -> Dict[str, int]:
+        """Gets the label distribution of the samples.
+
+        Returns:
+            label_distribution: a dict mapping label to count.
+        """
+        label_distribution = {}
+        for sample in self.samples:
+            if type(sample["label"]) == list:
+                for label in sample["label"]:
+                    label_distribution.setdefault(label, 0)
+                    label_distribution[label] += 1
+            else:
+                label_distribution.setdefault(sample["label"], 0)
+                label_distribution[sample["label"]] += 1
+        return label_distribution
+
+    def __getitem__(self, index) -> Dict:
+        """Returns a sample by index.
+
+        Returns:
+             Dict, a dict with patient_id, visit_id, and other task-specific
+                attributes as key. Conversion to index/tensor will be done
+                in the model.
+        """
+        return self.samples[index]
+
+    def __str__(self):
+        """Prints some information of the dataset."""
+        return f"{self.dataset_name} base dataset"
+
+    def __len__(self):
+        """Returns the number of samples in the dataset."""
+        return len(self.samples)
+
+    def stat(self) -> None:
+        """Prints some statistics of the dataset."""
+        self.task_stat()
+
+    # TODO: check this
+    def task_stat(self) -> None:
```

Comment:
Can you update this function to match BaseDataset.task_stat()? I think the get_label_distribution can be removed.